Modules/_testcapimodule.c

```diff
@@ -5169,6 +5169,226 @@ test_tstate_capi(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+// Test dict watching
```

Comment:
Would it be possible to wrap the dispatch machinery here and move the actual tests to the Python module?